Sources/AsyncQueue/AsyncQueue.swift

```diff
@@ -20,4 +20,70 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-final class AsyncQueue {}
+/// A queue that enables ordered sending of events from synchronous to asynchronous code
+public final class AsyncQueue: Sendable {
+
+    // MARK: Initialization
+
+    /// Instantiates an asynchronous queue.
+    /// - Parameter priority: The priority of the tasks added to the asynchronous queue.
+    public init(priority: TaskPriority = .medium) {
+        var capturedTaskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation? = nil
+        let taskStream = AsyncStream<@Sendable () async -> Void> { continuation in
+            capturedTaskStreamContinuation = continuation
+        }
+        taskStreamContinuation = capturedTaskStreamContinuation
+
+        streamTask = Task.detached(priority: priority) {
+            for await task in taskStream {
+                await task()
+            }
+        }
+    }
+
+    deinit {
+        taskStreamContinuation.finish()
+    }
+
+    // MARK: Public
+
+    /// Schedules an asynchronous task for execution and immediately returns.
+    /// The schedueled task will not execute until all prior tasks have completed.
+    /// - Parameter task: The task to enqueue.
+    public func async(_ task: @escaping @Sendable () async -> Void) {
+        taskStreamContinuation.yield(task)
+    }
+
+    /// Schedules an asynchronous throwing task and returns after the task is complete.
+    /// The schedueled task will not execute until all prior tasks have completed.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async -> T) async -> T {
+        await withUnsafeContinuation { continuation in
+            taskStreamContinuation.yield {
+                continuation.resume(returning: await task())
+            }
+        }
+    }
+
+    /// Schedules an asynchronous task and returns after the task is complete.
+    /// The schedueled task will not execute until all prior tasks have completed.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async throws -> T) async throws -> T {
+        try await withUnsafeThrowingContinuation { continuation in
+            taskStreamContinuation.yield {
+                do {
+                    continuation.resume(returning: try await task())
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+
+    // MARK: Private
+
+    private let streamTask: Task<Void, Never>
+    private let taskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation!
```

Comment:
Not ideal that we're force-unwrapping here. I originally made this type optional, but figured we really do want to fail LOUDLY if somehow we don't have a `taskStreamContinuation` and are attempting to append tasks to the queue.

Reply:
Cane we do the failing at the init method so we can initialize this with a non-optional?

Reply:
We can, but that would force every call-site to force unwrap, which seems pretty gnarly. Since `taskStreamContinuation` should exist (and it not existing would mean that the world is very broken), I think it's reasonable to crash if it somehow does not exist.

Reply:
Oh, I meant failing in a very general case. Crashing is fine for something like this where the logic means we would _never_ expect a `nil` value.

Reply:
The point was to remove the exclamation mark from here (together with the changes in `init` proposed).

Reply:
How's this look? 6e0519c

Reply:
This is silly, but I'm going to use a force-unwrap and a comment instead of a `fatalError`. I want 100% coverage 😅